### DISCUSSION TITLE
Add Custom commands

### DISCUSSION THREAD

**Discussion by cleder** [sqlalchemy/alembic#1456] [Usage Questions] [answered]
I'd like to add a custom command to alembic that produces a data migration.

Gemini and chatGPT hallucinate something about:
>  Create a custom Alembic command by subclassing Command from Alembic's commands module. 

so this may have been possible with a previous version.

**Comment by zzzeek** [comment, marked as answer]
sure, it's CommandLine and it's right here:  https://github.com/sqlalchemy/alembic/blob/main/alembic/config.py#L347  source code is straightforward

**Comment by cleder** [comment]
Sure, I can maintain a fork the question was more if it is possible with a plugin like system.

**Reply by CaselIT** [reply]
Not right now. I'm not sure if there could be something simple that we could support.
You could likely subclass CommandLine to add the new option and generate a script that uses the new class.

Your would not need to maintain a fork, but I think another command name other than `alembic` would be needed

**Reply by mikeroll** [reply]
My proposal for improving such a use case: https://github.com/sqlalchemy/alembic/pull/1611
